editoast/src/views/timetable.rs

```diff
@@ -269,6 +278,50 @@ async fn train_schedule(
     Ok(Json(train_schedule.into_iter().map_into().collect()))
 }
 
+/// Create paced trains by batch
+#[utoipa::path(
+    post, path = "",
+    tag = "timetable,paced_train",
+    params(TimetableIdParam),
+    request_body = Vec<PacedTrainBase>,
+    responses(
+        (status = 200, description = "The created paced trains", body = Vec<PacedTrainResult>)
+    )
+)]
+async fn paced_train(
+    State(db_pool): State<DbConnectionPoolV2>,
+    Extension(auth): AuthenticationExt,
+    Path(TimetableIdParam { id: timetable_id }): Path<TimetableIdParam>,
+    Json(paced_trains): Json<Vec<PacedTrainBase>>,
+) -> Result<Json<Vec<PacedTrainResult>>> {
+    let authorized = auth
+        .check_roles([BuiltinRole::TimetableWrite].into())
+        .await
+        .map_err(AuthorizationError::AuthError)?;
+    if !authorized {
+        return Err(AuthorizationError::Forbidden.into());
+    }
+
+    let conn = &mut db_pool.get().await?;
+
+    TimetableWithTrains::retrieve_or_fail(conn, timetable_id, || TimetableError::NotFound {
+        timetable_id,
+    })
+    .await?;
+    let changesets: Vec<PacedTrainChangeset> = paced_trains
+        .into_iter()
+        .map(|pt| PacedTrainForm {
+            timetable_id: Some(timetable_id),
+            paced_train: pt,
+        })
+        .map_into()
+        .collect();
```

Comment:
I'm not sure I agree with the existence of `struct PacedTrainForm`. With the suffix `Form`, I expect a payload that the `front` is sending. But in this case, the `front` creates paced trains with a `PacedTrainBase` on an endpoint that have a timetable identifier: `PacedTrainForm` is only constructed from these only on the backend.

Instead, maybe you can just do the following?

```suggestion
    let changesets = paced_trains
        .into_iter()
        .map(PacedTrainChangeset::from)
        .map(|cs| cs.timetable_id(timetable_id))
        .collect::<Vec<_>>();
```

Then you should be able to remove the `struct PacedTrainForm` and all related code.

Reply:
Nice catch :)